src/dbnode/storage/result.go

```diff
@@ -32,20 +34,79 @@ type tickResult struct {
 	mergedOutOfOrderBlocks int
 	errors                 int
 	evictedBuckets         int
+	metricToCardinality    map[string]int
```

Comment:
This struct is unbounded, can we set a bound like 10k here. I used to fixed a leaked metric issue which had 250k metric names: https://databricks.atlassian.net/browse/PLAT-55788

Reply:
I'll do that, though it's not trivial because the map is used to count the cardinality of each metric by incrementing the counter +1. Only at the end, is the map truncated to top 20.

Reply:
Done